### PR TITLE
Add CI through GitHub Actions

### PR DESCRIPTION
**Pull request by fdiaz**
This adds a simple CI through GitHub Actions

### CODE REVIEW

**Review by GeorgeLyon — APPROVED** [review]
Great, thanks for doing this! Can this work for Linux as well? (I made sure the tests worked :-)

**Comment by fdiaz** [comment]
@GeorgeLyon we can definitively add another setup for linux! Let's merge this one first and add another one on top of this to make sure this continues running on Linux if that's a goal of this project.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    runs-on: macos-latest
+    steps:
+    # Checks-out the repo. More at: https://github.com/actions/checkout
+    - uses: actions/checkout@v2
+    - name: Select Xcode version
+      run: sudo xcode-select -switch /Applications/Xcode_11.4.app
+    - name: Build project
+      run: swift build
+    - name: Run tests
+      run: swift test
```
Comment (dfed) [inline .github/workflows/ci.yml:16]: What is the oldest OS this project supports? We should be running tests against that SDK if possible.

Comment (fdiaz) [reply .github/workflows/ci.yml:16]: @dfed We only get macOS 15 from GitHub actions, so I think running on the latest versions makes sense here https://help.github.com/en/actions/reference/software-installed-on-github-hosted-runners

Comment (dfed) [reply .github/workflows/ci.yml:16]: Got it. @fdiaz can we use GitHub actions to build on an iOS simulator? Or do we need a 3rd party integration for that?

Comment (fdiaz) [reply .github/workflows/ci.yml:16]: We have control on what command we run and we have Xcode installed, so we should be able to build using `xcodebuild`

**Pull request merged**